### PR TITLE
Bug fixes for TTO

### PR DESCRIPTION
**Pull request by wenatuhs**
### Bug fixes

- Fix the cannot-delete-last-run issue
- Support constraints/states in routine editor
- Auto-select the updated routine in the routine list once the changes are saved

### Remaining issues

- Critical constraints being triggered incorrectly (Xopt-related)
- Possible faulty behavior when add constraints in routine

### CODE REVIEW

**Review by roussel-ryan — CHANGES_REQUESTED** [review]
Mostly looks good, some minor comments to address

File: src/badger/gui/default/components/routine_page.py [hunk 1]
```diff
@@ -547,19 +547,19 @@ def _compose_vocs(self) -> (VOCS, List[str]):
             if critical:
                 critical_constraints.append(con_name)
 
-        states = {}
+        states = []
```
[the remark this answers is not in the document]
Comment (wenatuhs) [reply src/badger/gui/default/components/routine_page.py:550]: Sure thing! They mean the same, the concern was that the observables here get confused with the observables in Badger env, but maybe I'm overthinking

File: src/badger/gui/default/components/run_monitor.py [hunk 2]
```diff
@@ -151,7 +151,10 @@ def __init__(self):
 
     @property
     def vocs(self) -> VOCS:
-        return self.routine.vocs
+        if self.routine is None:
+            return None
```
[the remark this answers is not in the document]
Comment (wenatuhs) [reply src/badger/gui/default/components/run_monitor.py:155]: It's a valid status that the run monitor doesn't hold a routine -- say when the user just gets started and creates a few routines without running them, in this case, if the user doesn’t select any routine in the list the run monitor should be in this empty status

File: src/badger/gui/default/components/run_monitor.py [hunk 3]
```diff
@@ -477,18 +483,14 @@ def init_plots(self, routine: Routine = None, run_filename: str = None):
                 self.plot_sta
             except:
                 self.plot_sta = plot_sta = add_axes(
-                    self.monitor, "Constants", 'Evaluation History (S)',
-                    self.inspector_state, row=1, col=0
+                    self.monitor, "states", 'Evaluation History (S)',
```
[the remark this answers is not in the document]
Comment (wenatuhs) [reply src/badger/gui/default/components/run_monitor.py:486]: Will change to `plot_obs`!

**Review by roussel-ryan — APPROVED** [review]

**Pull request merged**